service.rb

```diff
@@ -33,7 +34,14 @@ def initialize(port)
         end
 
         response.redirect('/show/data', status: 303)
-      
+
+      elsif request.get? && request.path == "/daily_data.json"
+        response.content_type = "text/plain"
+        all_data = {}
```

Comment:
FYI the `store.transaction` call returns the value of its block, so instead of initialising `all_data` outside the block and assigning to it inside, you could just have the block return the value you’re interested in and assign it once:

```ruby
all_data = store.transaction { store[:all_data] }
```